base_layer/core/src/proof_of_work/difficulty.rs

```diff
@@ -89,15 +121,54 @@ impl fmt::Display for Difficulty {
     }
 }
 
-impl From<u64> for Difficulty {
-    fn from(value: u64) -> Self {
-        Difficulty(value)
+#[cfg(any(test))]
+use std::ops::{Add, Div, Mul, Sub};
+
+// This trait should not be implemented for runtime, here only for testing
+#[cfg(any(test))]
+impl Add<Self> for Difficulty {
+    type Output = Self;
+
+    fn add(self, _rhs: Self) -> Self {
+        unimplemented!("Sub::sub<rhs> must not be used, use `.checked_add(value)` instead")
```

Comment:
I am not the biggest fan of this. I understand the reason here is to stop people from implementing the methods. 
My personal opinion here is that, I don't think the slight chance of that happening in getting missed in a PR is worth the extra clutter to keep these non-functioning methods in the file.